thirdparty/breakpad/mainwindow.cpp

```diff
@@ -0,0 +1,334 @@
+
+    //DWORD pid = GetCurrentProcessId();
+
+    // Open a Windows Process equivelant to fork() for Linux
+    if( !CreateProcess( NULL,   // No module name (use command line)
+            (WCHAR *)mycmd.c_str(),          // Command line
+            NULL,           // Process handle not inheritable
+            NULL,           // Thread handle not inheritable
+            FALSE,          // Set handle inheritance to FALSE
+            0,              // No creation flags
+            NULL,           // Use parent's environment block
+            NULL,           // Use parent's starting directory
+            &si,            // Pointer to STARTUPINFO structure
+            &pi )           // Pointer to PROCESS_INFORMATION structure
+        )
+        {
+            printf( "CreateProcess failed (%lu).\n", GetLastError() );
+            return true;
+        }
+
+    // Wait until child process exits.
+    //WaitForSingleObject( pi.hProcess, INFINITE );
+
+    // Close process and thread handles.
+
+    //exit(0);
+    //CloseHandle( pi.hProcess );
+    //CloseHandle( pi.hThread );
+
+
+    Q_UNUSED(program);
+    return false;
+
+}
+
+QString get_crashreporter_path(){
+    wchar_t buffer[MAX_PATH];
+    GetModuleFileName(NULL, buffer, MAX_PATH) ;
+    string mscore_path = wstr2str(wstring(buffer));
+
+    int i;
+    int path_n;
+
+    for(i=mscore_path.size();i>=0;i--){
+        if( mscore_path[i] == '\\'){
+            path_n = i;
+            break;
+        }
+    }
+
+    string res;
+
+    for(i=0;i<path_n;i++){
+        res += mscore_path[i] ;
+    }
+
+    return QString(res.c_str());
+}
+
+
+QString get_musescore_path(){
+    QString crashreporter_path = get_crashreporter_path();
+    QString res = crashreporter_path+"\\MuseScore.exe";
+    QString res_empty = "";
+    QFileInfo fileInfo(res);
+    if ( fileInfo.exists() && fileInfo.isFile()){
+        return res;
+    }
+
+    res = crashreporter_path + "\\..\\..\\mscore\\MuseScore.exe";
+    QFileInfo fileInfo2(res);
+    if ( fileInfo2.exists() && fileInfo2.isFile()){
+        return res;
+    }
+
+    return res_empty;
+
+}
+
+wstring str2wstr(string mystr){
+    wstring res(mystr.begin(), mystr.end());
+    return res;
+}
+
+string wstr2str(wstring mystr){
+    string res(mystr.begin(), mystr.end());
+    return res;
+}
+
+pair<string,string> line2strings(string line){
+    string str1, str2;
+    string *mystr;
+    bool myflag;
+    int i;
+
+    myflag = false;
+    str1.empty();
+    str2.empty();
+
+    mystr = &str1;
+
+    for (i = 0; i < line.size(); i++){
+            if ( line[i] == ',' && myflag == false){
+                    mystr = &str2;
+                    myflag = true;
+            }
+            else{
+                    if (line[i] != '\n'){
+                            *mystr += line[i];
+                    }
+
+            }
+    }
+
+    return pair<string,string>(str1,str2);
+
+}
+
+QMap <QString,QString> read_csv(QString mypath){
```

Comment:
this is more than just reading a random CSV. Give it a descriptive name and add some comments before the function.